GPU/Common/FragmentShaderGenerator.cpp

```diff
@@ -947,12 +954,13 @@ bool GenerateFragmentShader(const FShaderID &id, char *buffer, const ShaderLangu
 						WRITE(p, "  if ((colortest.r %s u_alphacolorref.r) && (colortest.g %s u_alphacolorref.g) && (colortest.b %s u_alphacolorref.b)) %s\n", test, test, test, discardStatement);
 					} else if (compat.bitwiseOps) {
 						WRITE(p, "  ivec3 v_scaled = roundAndScaleTo255iv(v.rgb);\n");
+						WRITE(p, "  uvec3 colormask = unpackIVec3(u_alphacolormask);\n");
 						if (compat.shaderLanguage == GLSL_VULKAN) {
 							// Apparently GLES3 does not support vector bitwise ops, but Vulkan does?
-							WRITE(p, "  if ((v_scaled & u_alphacolormask.rgb) %s (u_alphacolorref.rgb & u_alphacolormask.rgb)) %s\n", colorTestFuncs[colorTestFunc], discardStatement);
+							WRITE(p, "  if ((v_scaled & colormask) %s (u_alphacolorref.rgb & colormask)) %s\n", colorTestFuncs[colorTestFunc], discardStatement);
```

Comment:
I wonder if it'd be better in both these cases to replace `roundAndScaleTo255iv` with just a pack to unorm, and then mask the entire uint.

-[Unknown]

Reply:
Well, there's the issue of the compares, right, that operate per component? Or hm, did they actually, or was that just a bug workaround we did on d3d11?

Reply:
No there was definitely an issue with comparing a vec.  I'm saying, instead of:

```
ivec3 v_scaled = floor and scale v to 255;
uvec3 colormask = convert mask to vec
compare r, g, b
```

We do:
```
uint v_scaled = packUnorm4x8(v);
compare all at once (no components)
```

-[Unknown]

Reply:
Right, this would work. We need to pack alphacolorref as well for that, I kinda wanted to "pre-pack" it, but it would be incompatible with the OpenGL stuff below. I might do it anyway in a followup, for now I'm leaving it like it is, I think.

Reply:
Added that basically in #16109.

-[Unknown]